static/src/javascripts/projects/commercial/modules/dfp/Advert.ts

```diff
@@ -116,11 +116,14 @@ class Advert {
 
 		this.sizes = this.generateSizeMapping(additionalSizeMapping);
 
```

Comment:
Throw an error if we cannot define an ad slot. Catch the error where the advert was created. I assumed that we don't want to create an Advert object if we cannot successfully define a slot for it.